testing/sdk_auth.py

```diff
@@ -0,0 +1,376 @@
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
+                log.info("Task id is {}".format(task["id"]))
+                return task["id"]
+
+    raise RuntimeError("Expecting marathon KDC task but no such task found. Running tasks: {tasks}".format(
+        tasks=raw_tasks))
+
+
+def _get_framework_id() -> str:
+    """
+    :return (str): The ID of the framework running the KDC app.
+    """
+    log.info("Getting framework id")
+    raw_tasks = sdk_cmd.run_cli("task --json")
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
+                log.info("Framework id is {}".format(task["framework_id"]))
+                return task["framework_id"]
+
+    raise RuntimeError("Expecting marathon KDC task but no such task found. Running tasks: {tasks}".format(
+        tasks=raw_tasks))
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_host_id() -> str:
+    """
+    :return (str): The ID of the node running the KDC app.
+    """
+    log.info("Getting host id")
+    raw_tasks = sdk_cmd.run_cli("task --json")
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
+                log.info("Host id is {host_id}".format(host_id=task["slave_id"]))
+                return task["slave_id"]
+
+    raise RuntimeError("Expecting marathon KDC task but no such task found. Running tasks: {tasks}".format(
+                        tasks=raw_tasks))
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_host_name(host_id: str) -> str:
+    """
+    Fetches the host name for the host running the KDC app.
+    :param host_id (str): The ID of the host, used to look up the appropriate node.
+    :return (str): Name of the host running the KDC app.
+    """
+    log.info("Getting host name")
+    raw_nodes = sdk_cmd.run_cli("node --json")
+    if raw_nodes:
+        nodes = json.loads(raw_nodes)
+        for node in nodes:
+            if node["id"] == host_id:
+                log.info("Host name is {host_name}".format(host_name=node["hostname"]))
+                return node["hostname"]
+
+    raise RuntimeError("Failed to get name of host running the KDC app: {nodes}")
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_master_public_ip() -> str:
+    """
+    :return (str): The public IP of the master node in the DC/OS cluster.
+    """
+    dcos_url, headers = sdk_security.get_dcos_credentials()
+    cluster_metadata_url = "{cluster_url}/metadata".format(cluster_url=dcos_url)
+    raw_response = sdk_cmd.request("GET", cluster_metadata_url, verify=False)
+    if not raw_response.ok:
+        raise RuntimeError("Unable to get the master node's public IP address: {err}".format(err=repr(raw_response)))
+
+    response = json.loads(raw_response.text)
```

Comment:
Is `raw_response` a wrapper around a `request`  response object? If so, then it may have a `.json()` method that one could use directly. (as a future note)

Reply:
Yup, eradicated wrapper.